### DISCUSSION TITLE
Incorrect sorting on expense accounts

### DISCUSSION THREAD

**Discussion by pc-zookeeper** [firefly-iii/firefly-iii#9080] [Firefly III] [answered]
I've been using and enjoying firefly-iii since v4.7.2 and currently self-host v6.1.18 today via docker and MariaDB.  Thank you for continuing to develop and support such a great application.

I've been having the same issue as reported in [#4174](https://github.com/firefly-iii/firefly-iii/issues/4174) since the v5.4.6 upgrade.  In the issue report, a community member was kind enough to post a PostgreSQL script to perform a correction of the expense account sorting issue.  Over the past few years here and there, I've attempted to modify the script for MariaDB and have failed to get it to work correctly each time, resulting in having to restore the backup I had taken prior to attempting the fix.

It might be a super simple fix, but code and I don't get along, and with 500+ expense accounts, sorting by hand is not ideal with the amount of time it would consume.  Would someone be able to assist me in converting the posted PostgreSQL to something that will work within MariaDB?

In case it matters, I also have a relative using my instance with their own user account.  They are unaffected by this issue as they started using firefly-iii only a year ago.  Of course, I don't want to cause their data to become entangled within my own.

If someone could help me out, I'd greatly appreciate it!

**Comment by JC5** [comment]
I can't replicate this issue myself though. Where is the sorting incorrect?

**Comment by pc-zookeeper** [comment]
Its within the Expense Accounts list, dashboard>accounts>expense accounts.

Here are 3 screen captures from different pages within the list:
![image](https://github.com/user-attachments/assets/31df345f-f0fd-43a2-b73d-1222f469efef)
![image](https://github.com/user-attachments/assets/7a6e71f3-b65f-46e9-a3aa-87c946e3f62f)
![image](https://github.com/user-attachments/assets/81772c67-1d9d-4ad5-8388-fc8cf6f25d4d)

**Reply by JC5** [reply]
Weird. On the demo site, I can't get other orders to stick.

**Reply by pc-zookeeper** [reply]
I can move the accounts around manually via drag and drop and they'll stay in the order I leave them in (as far as I can tell anyhow).  That being said, manual sorting is an option, just an extremely time consuming one which is why I'm hoping someone can provide a SQL script (MariaDB specific) to fix this in a couple seconds from within the DB.

**Reply by JC5** [reply]
Yes, but like the demo site, they should snap back in place after you refresh the page. There's a specific routine in place to sort them alphabetically. I'll check out the code.

**Reply by pc-zookeeper** [reply]
I confirm that after manually changing the order of a couple accounts and refreshing the page, they did snap back to their old spots.

Used the 2 identified accounts
![image](https://github.com/user-attachments/assets/dbe68ccc-0784-4893-8456-8a834d3c8ca6)

Dragged the 7 over the 2
![image](https://github.com/user-attachments/assets/a30fa775-6581-483a-9780-1b6468e021aa)

Refreshed the page
![image](https://github.com/user-attachments/assets/1312fc66-fcb7-483c-a09e-ff63adb26e7e)

For reference, this is on page 6 of 12 of my expense account list.

**Reply by JC5** [reply, marked as answer]
I see what you mean. You could run the following query to fix it. The next release will do this automatically:

```sql
update `accounts` left join `account_types` on `account_types`.`id` = `accounts`.`account_type_id` set `order` = 0 where  `accounts`.`user_id` is not null and `account_types`.`type` not in ('Default account', 'Asset account', 'Loan', 'Debt', 'Credit card', 'Mortgage') and `accounts`.`deleted_at` is null
```

**Reply by pc-zookeeper** [reply]
This fixed it, thank you so much!

**Reply by JC5** [reply]
Hah, thank you for your support! 🎉
